op-e2e/actions/upgrades/interop_fork_test.go

```diff
@@ -0,0 +1,144 @@
+	require.False(t, rollupConfigA.IsInterop(currentBlockHeaderA.Time),
+		"Interop should not be active at the first L1 inclusion block in chain A")
+	require.False(t, rollupConfigB.IsInterop(currentBlockHeaderB.Time),
+		"Interop should not be active at the first L1 inclusion block in chain B")
+
+	// Build a few L2 blocks. We only need the L1 inclusion to advance past Interop and Interop
+	// shouldn't activate with L2 time.
+	actors.ChainA.Sequencer.ActBuildL2ToInterop(t)
+	actors.ChainB.Sequencer.ActBuildL2ToInterop(t)
+
+	////////////////////////////
+	// Post-upgrade Block Production
+	////////////////////////////
+
+	activationBlockHeaderA := actors.ChainA.SequencerEngine.L2Chain().CurrentBlock()
+	activationBlockHeaderB := actors.ChainB.SequencerEngine.L2Chain().CurrentBlock()
+	require.Equal(t, activationBlockHeaderA.Number.Uint64(), activationBlockHeaderB.Number.Uint64())
+	activationBlockNumber := activationBlockHeaderA.Number
+
+	require.True(t, rollupConfigA.IsInteropActivationBlock(activationBlockHeaderA.Time),
+		"Interop should be active at the first L1 inclusion block in chain A")
+	require.True(t, rollupConfigB.IsInteropActivationBlock(activationBlockHeaderB.Time),
+		"Interop should be active at the first L1 inclusion block in chain B")
+
+	activationBlock := actors.ChainA.SequencerEngine.L2Chain().GetBlockByHash(activationBlockHeaderA.Hash())
+	activationBlockTxs := activationBlock.Transactions()
+	VerifyInteropContractsDeployedCorrectly(t, actors, activationBlockTxs, activationBlockNumber)
+
+	actors.ActBatchAndMine(t, actors.ChainA, actors.ChainB)
+
+	actors.ChainA.Sequencer.ActL2PipelineFull(t)
+	actors.ChainB.Sequencer.ActL2PipelineFull(t)
+
+	// The node will exhaust L1 data,
+	// it needs the supervisor to see the L1 block first,
+	// and provide it to the node.
+	for i := 0; i < 2; i++ {
+		actors.ChainA.Sequencer.ActL2EventsUntil(t, event.Is[derive.ExhaustedL1Event], 100, false)
+		actors.ChainB.Sequencer.ActL2EventsUntil(t, event.Is[derive.ExhaustedL1Event], 100, false)
+		actors.Supervisor.SignalLatestL1(t)          // supervisor will be aware of latest L1
+		actors.ChainA.Sequencer.SyncSupervisor(t)    // supervisor to react to exhaust-L1
+		actors.ChainB.Sequencer.SyncSupervisor(t)    // supervisor to react to exhaust-L1
+		actors.ChainA.Sequencer.ActL2PipelineFull(t) // node to complete syncing to L1 head.
+		actors.ChainB.Sequencer.ActL2PipelineFull(t) // node to complete syncing to L1 head.
+	}
+
+	actors.ChainA.Sequencer.ActL1HeadSignal(t)
+	actors.ChainB.Sequencer.ActL1HeadSignal(t)
+
+	actors.ChainA.Sequencer.SyncSupervisor(t) // supervisor to react to exhaust-L1
+	actors.ChainB.Sequencer.SyncSupervisor(t) // supervisor to react to exhaust-L1
+
+	actors.Supervisor.ProcessFull(t)
+
+	actors.ChainA.Sequencer.ActL2PipelineFull(t) // node to complete syncing to L1 head.
+	actors.ChainB.Sequencer.ActL2PipelineFull(t) // node to complete syncing to L1 head.
```

Comment:
Some of this seems redundant. We should identify in the follow-up what are the minimal steps to make an interop system aware of new L1 data and derive up to the latest L1 block. It shouldn't be that many actions.

Reply:
We should then also put this into a helper method on `actors` so it can easily be reused in other interop action tests.